server/app/routes/v1/grids_api.rb

```diff
@@ -127,8 +127,16 @@ class GridsApi < Roda
             end
             unless r['services'].nil?
               services = r['services'].split(',').map do |service|
-                @grid.grid_services.find_by(name: service).try(:id)
-              end.delete_if{|s| s.nil?}
+                if service.include?('/')
+                  stack_name, service_name = service.split('/', 2)
+                  stack_id = @grid.stacks.where(name: stack_name).first.try(:id)
+                  if stack_id && service_name
+                    @grid.grid_services.where(stack_id: stack_id, name: service_name).first.try(:id)
+                  end
+                else
+                  @grid.grid_services.find_by(name: service).try(:id)
```

Comment:
That's also an option, because at the API level, looking at how the `/v1/services` URLs work, the `redis` case should actually be `/v1/grids/:id/container_logs?service=null/redis` (with the explicit `null` stack prefix).

However, we don't want to expose those in the CLI, so `kontena grid logs --service null/redis` should be avoided, and `kontena grid logs --service redis` should IMO be normalized to `null/redis`, like it is for the `kontena service` commands...